packages/remirror__extension-code-block/src/code-block-extension.ts

```diff
@@ -293,6 +293,10 @@ export class CodeBlockExtension extends NodeExtension<CodeBlockOptions> {
    * Format the code block with the code formatting function passed as an
    * option.
    *
+   * The formatter runs asynchronously. This should not be immediately followed
+   * by additional commands. It is not yet possible to track completion of this
+   * command; it is best to use as a standalone response to a user action.
+   *
```

Comment:
If this is an async command, we should use the `DelayedCommand` interface instead, as this supports promises.

There's an example of [its usage here](https://github.com/remirror/remirror/blob/a0cc38df84d3de7d3fd72ae1804803bf4b76b0de/packages/remirror__core/src/builtins/decorations-extension.ts#L599)

This should mean the command is chainable, but if that's not the case we could alternatively

```ts
@command({
  disableChaining: true,
})
```

Reply:
Thanks for the example. I followed it pretty closely for the changes in [5a498ef](https://github.com/remirror/remirror/pull/2255/commits/5a498ef8009d19c7e0a22d678f5ec97e1ad8a0f8). I'm not familiar with best/worst practices in this project, so I'm curious about whether it's ok to pass an `EditorState` to the promise as I've done in this commit.

While working on these changes, I think I may have fixed a bug [here](https://github.com/remirror/remirror/blob/08e0da14d48f144e8be1c43fea398cfad8f518e7/packages/remirror__extension-code-block/src/code-block-utils.ts#L276-L279). A `pos` parameter is allowed to be passed into the format function to identify a different code block than currently selected, but then the current selection was used to find the code block anyways (i.e. `pos` was ignored). Let me know if you agree with the change I made [here](https://github.com/remirror/remirror/pull/2255/commits/5a498ef8009d19c7e0a22d678f5ec97e1ad8a0f8#diff-72c26535e24a6bdefe72fe7bc93b77f5bf07518fb41d15d8586ce67b7abc4909R293-R296) to utilize `pos` if provided.

Finally, my changes in `packages/remirror__extension-code-block/__tests__/code-block-extension.spec.ts` remain because I'm not sure how else to wait for the command to signal completion before checking the updated doc. Let me know if there's a mechanism for this.